sql/core/src/main/scala/org/apache/spark/sql/DataFrameImpl.scala

```diff
@@ -409,8 +411,26 @@ private[sql] class DataFrameImpl protected[sql](
   override def toJSON: RDD[String] = {
     val rowSchema = this.schema
     this.mapPartitions { iter =>
-      val jsonFactory = new JsonFactory()
-      iter.map(JsonRDD.rowToJSON(rowSchema, jsonFactory))
+      val writer = new CharArrayWriter()
```

Comment:
I was trying to use the `StringWriter`, but it doesn't have the method like `reset`, hence we may not able to reuse the object for for each of the records. That's why I changed it as `CharArrayWriter`.